android/src/main/java/com/twiliovoicereactnative/CallMessageListenerProxy.java

```diff
@@ -54,24 +63,31 @@ public void onMessageSent(String voiceEventSID) {
   }
 
   @Override
-  public void onMessageReceived(CallMessage callMessage) {
+  public void onMessageReceived(String callSid, CallMessage callMessage) {
     logger.debug("onMessageReceived");
 
-    // notify JS layer ScopeCall
-    getJSEventEmitter().sendEvent(ScopeCall,
-      constructJSMap(
-        new Pair<>(VoiceEventType, CallEventMessageReceived),
-        new Pair<>(JSEventKeyCallMessageInfo, serializeCallMessage(callMessage))
-      )
-    );
+    //final call record
+    final CallRecord callRecord =
+      Objects.requireNonNull(getCallRecordDatabase().get(new CallRecord(callSid)));
 
-    // notify JS layer ScopeCallInvite
-    getJSEventEmitter().sendEvent(ScopeCallInvite,
-      constructJSMap(
-        new Pair<>(VoiceEventType, CallEventMessageReceived),
-        new Pair<>(JSEventKeyCallMessageInfo, serializeCallMessage(callMessage))
-      )
-    );
+    if (CallRecord.CallInviteState.ACTIVE == callRecord.getCallInviteState()) {
+      // notify JS layer ScopeCallInvite
+      getJSEventEmitter().sendEvent(ScopeCallInvite,
+        constructJSMap(
+          new Pair<>(VoiceEventType, CallEventMessageReceived),
+          new Pair<>(JSEventKeyCallMessageInfo, serializeCallMessage(callMessage))
+        )
+      );
+    } else {
+      // notify JS layer ScopeCall
```

Comment:
This if-else block is not needed, because it drops null values from the entry table, ie, this could be collapsed into..

```
      getJSEventEmitter().sendEvent(ScopeCallInvite,
        constructJSMap(
          new Pair<>(VoiceEventType, CallEventMessageReceived),
          new Pair<>(JS_EVENT_KEY_CALL_INFO, (CallRecord.CallInviteState.ACTIVE == callRecord.getCallInviteState()) ? serializeCall(callRecord), null),
          new Pair<>(JSEventKeyCallMessageInfo, serializeCallMessage(callMessage))
        )
      );
```

Reply:
I also need a ternary for the first argument in `sendEvent`. Made reflecting change